oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java

```diff
@@ -125,6 +127,29 @@ static class AwsCredentialSource extends CredentialSource {
       } else {
         this.imdsv2SessionTokenUrl = null;
       }
+
+      this.validateMetadataServerUrls();
+    }
+
+    private void validateMetadataServerUrls() {
+      this.validateMetadataServerUrlIfAny(this.regionUrl, "region_url");
+      this.validateMetadataServerUrlIfAny(this.url, "url");
+      this.validateMetadataServerUrlIfAny(this.imdsv2SessionTokenUrl, "imdsv2_session_token_url");
+    }
+
+    private void validateMetadataServerUrlIfAny(String urlString, String nameOfData) {
+      if (urlString != null) {
```

Comment:
add the "|| urlString.trim().length() == 0" for empty strings? or we consider it an attempt to set a correct url?

Reply:
Empty url should be ok. If an environment variable already provides the necessary info, these urls will not be used. Later in the code we check to see if both env var and url are not provided, if so we throw an appropriate exception. Here we just make sure that a unknown host is not provided.